### DISCUSSION TITLE
How to call `fastlane pilot add...` from Fastfile

### DISCUSSION THREAD

**Discussion by CraigSiemens** [fastlane/fastlane#20402] [Q&A]
The `pilot` method that can be called from a Fastfile doesn't have an obvious option to add a tester the same way that that can be done by running the command directly. The command has a subcommand that is called but methods don't have something similar.

I'm trying to figure out what options to pass to pilot in a fast file to reproduce a call similar to the cli example in the docs.

```bash
fastlane pilot add email@email.com -a com.krausefx.app -g group-1,group-2
```
https://docs.fastlane.tools/actions/pilot/#pilot

I tried a command like this only setting the options needed to add a tester but it fails.

```ruby
pilot(
  app_identifier: identifier,
  first_name: first_name,
  last_name: last_name,
  email: email,
  groups: ["My Group"]
)
```

```bash
ERROR [2022-06-21 11:00:34.73]: No ipa or pkg file given
```

**Comment by lucgrabowski** [comment]
You could wrap the command in `sh` and call lane with options passed to `sh` command. Just to give you an example:

```ruby
lane :add_tester do |options|
     sh("fastlane pilot add #{options[:email]} -a #{options[:app_identifier]} -g #{options[:groups]}"
end
```
You'll have to play around with `groups` parameter, it's possible that it will require escaping it with `\"`

**Comment by Palmieri78759** [comment]
The best way to manage your TestFlight testers and builds from your terminal
Pilot makes it easier to manage your app on Apple’s TestFlight. You can:

Upload & distribute builds
Add & remove testers
Retrieve information about testers & devices
Import/export all available testers
pilot uses spaceship.airforce to interact with App Store Connect 🚀

[PFM Driver Login](https://www.pfmlogin.net/)
